docs/release-notes/release-2024-06-10.md

```diff
@@ -0,0 +1,258 @@
+```
+
+### Array values
+
+Array values can be declared with the `#[]` syntax.
+
+```tsp
+const users = #["Bob", "Frank"];
+```
+
+### `const`
+
+Const are the value world equivalent of `alias` but also allows you to specify a type
+
+```tsp
+const name = "Bob";
+const name: string = "Bob";
+```
+
+### Scalar constructors
+
+Scalars also needs to be able to define values. For scalars that don't extend a base type like `string`, `number` or `boolean` which have literal syntax you can define a constructor.
+
+```tsp
+scalar ipV4 {
+  init fromInt(value: int32);
+}
+const ip: ipV4 = ipV4.fromInt(3232235776);
+```
+
+[See values doc for more info](https://typespec.io/docs/language-basics/values)
+
+## Deprecations
+
+### @typespec/compiler
+
+- [#3022](https://github.com/microsoft/typespec/pull/3022) Using a tuple type as a value is deprecated. Tuple types in contexts where values are expected must be updated to be array values instead. A codefix is provided to automatically convert tuple types into array values.
+
+```tsp
+model Test {
+  // Deprecated
+  values: string[] = ["a", "b", "c"];
+
+  // Correct
+  values: string[] = #["a", "b", "c"];
+}
+```
+
+- [#3022](https://github.com/microsoft/typespec/pull/3022) Using a model type as a value is deprecated. Model types in contexts where values are expected must be updated to be object values instead. A codefix is provided to automatically convert model types into object values.
+
+```tsp
+model Test {
+  // Deprecated
+  user: {
+    name: string;
+  } = {
+    name: "System";
+  };
+
+  // Correct
+  user: {
+    name: string;
+  } = #{ name: "System" };
+}
+```
+
+- [#3022](https://github.com/microsoft/typespec/pull/3022) Decorator API: Legacy marshalling logic
+
+With the introduction of values, the decorator marshalling behavior has changed in some cases. This behavior is opt-in by setting the `valueMarshalling` package flag to `"new"`, but will be the default behavior in future versions. It is strongly recommended to adopt this new behavior as soon as possible.
+
+Example:
+
+```tsp
+extern dec multipleOf(target: numeric | Reflection.ModelProperty, value: valueof numeric);
+```
+
+Will now emit a deprecated warning because `value` is of type `valueof string` which would marshall to `Numeric` under the new logic but as `number` previously.
+
+To opt-in you can add the following to your library js/ts files.
+
+```ts
+export const $flags = definePackageFlags({
+  decoratorArgMarshalling: "new",
+});
+```
+
+## Features
+
+### @typespec/compiler
+
+- [#3280](https://github.com/microsoft/typespec/pull/3280) Support completion for Model with extended properties
+
+  Example
+
+  ```tsp
+  model Device {
+    name: string;
+    description: string;
+  }
+
+  model Phone extends Device {
+    ┆
+  } | [name]
+    | [description]
+  ```
+
+- [#3280](https://github.com/microsoft/typespec/pull/3280) Support completion for object values and model expression properties.
```

Comment:
Given we are shipping values here, I don't think we need this?

Reply:
Well this applies to model too not just object values